codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -510,31 +493,11 @@ public void onUnknownFrame(ChannelHandlerContext ctx, byte frameType, int stream
             onUnknownFrame0(ctx, frameType, streamId, flags, payload);
         }
 
-        /**
-         * Indicates whether or not frames for the given stream should be ignored based on the state of the
-         * stream/connection.
-         */
-        private boolean shouldIgnoreFrame(Http2Stream stream, boolean allowResetSent) {
-            if (connection.goAwaySent() &&
-                    (stream == null || connection.remote().lastStreamCreated() <= stream.id())) {
-                // Frames from streams created after we sent a go-away should be ignored.
-                // Frames for the connection stream ID (i.e. 0) will always be allowed.
-                return true;
-            }
-
-            // Also ignore inbound frames after we sent a RST_STREAM frame.
-            return stream != null && !allowResetSent && stream.isResetSent();
-        }
-
-        /**
-         * Verifies that a GO_AWAY frame was not previously received from the remote endpoint. If it was, throws a
-         * connection error.
-         */
-        private void verifyGoAwayNotReceived() throws Http2Exception {
-            if (connection.goAwayReceived()) {
-                // Connection error.
-                throw connectionError(PROTOCOL_ERROR, "Received frames after receiving GO_AWAY");
-            }
+        private boolean streamCreatedAfterGoAwaySent(Http2Stream stream) {
+            int lastStreamId = connection().remote().lastKnownStream();
```

Comment:
We should consider a few things here.
1. Save a reference to connection()
2. Inline the lastKnownStream call...you may never have to call it if the first portion of the conditional fails.